packages/@react-native-windows/cli/src/runWindows/runWindows.ts

```diff
@@ -4,68 +4,67 @@
  * @format
  */
 
-import crypto from 'crypto';
 import fs from 'fs';
 import path from 'path';
-import {
-  Telemetry,
-  isMSFTInternal,
-  getDiskFreeSpace,
-  CodedError,
-  CodedErrorType,
-  CodedErrors,
-} from '@react-native-windows/telemetry';
+import {Telemetry, CodedError} from '@react-native-windows/telemetry';
 
 import * as build from './utils/build';
 import chalk from 'chalk';
 import * as deploy from './utils/deploy';
-import {newError, newInfo, newWarn} from './utils/commandWithProgress';
+import {
+  newError,
+  newInfo,
+  newWarn,
+  setExitProcessWithError,
+} from './utils/commandWithProgress';
+import {
+  getDefaultOptions,
+  startTelemetrySession,
+  endTelemetrySession,
+} from './utils/telemetryHelpers';
 import * as info from './utils/info';
 import MSBuildTools from './utils/msbuildtools';
 
 import {Command, Config} from '@react-native-community/cli-types';
 import {runWindowsOptions, RunWindowsOptions} from './runWindowsOptions';
 
-import {autoLinkCommand} from './utils/autolink';
-import {totalmem, cpus} from 'os';
-
-function setExitProcessWithError(
-  error: Error,
-  loggingWasEnabled?: boolean,
-): void {
-  if (loggingWasEnabled !== true) {
-    console.log(
-      `Re-run the command with ${chalk.bold('--logging')} for more information`,
-    );
-    if (Telemetry.client) {
-      console.log(
-        `Your session id was ${Telemetry.client.commonProperties.sessionId}`,
-      );
-    }
-  }
-  if (error instanceof CodedError) {
-    process.exitCode = CodedErrors[error.name as CodedErrorType];
-  } else {
-    process.exitCode = 1;
+import {autolinkWindowsInternal, AutoLinkOptions} from './utils/autolink';
+
+/**
+ * Sanitizes the given option for telemetery.
+ * @param key The key of the option.
+ * @param value The unsanitized value of the option.
+ * @returns The sanitized value of the option.
+ */
+function optionSanitizer(key: string, value: any): any {
+  switch (key) {
+    case 'root':
+    case 'target':
+    case 'sln':
+    case 'proj':
+    case 'buildLogDirectory':
+      return value === undefined ? false : true;
+    case 'msbuildprops':
+      return value === undefined ? 0 : value.split(',').length;
+    default:
+      return value === undefined ? false : value;
```

Comment:
All args must be captured, but yes, using `keyof` in the option sanitizers can make sure that devs adding new args will have to explicitly decide on sanitization behavior. I've also expanded the tests to make sure that devs must also add the args to the options interfaces in the first place (which is currently not enforced).

Reply:
I fixed this for run-windows and autolink-windows, because the interfaces exist. I will do react-native-windows-init in another PR. Tracking issue: https://github.com/microsoft/react-native-windows/issues/9163